packages/core/src/browser/shell/application-shell.ts

```diff
@@ -639,15 +639,22 @@ export class ApplicationShell extends Widget implements WidgetTracker {
             console.error('Widgets added to the application shell must have a unique id property.');
             return;
         }
+        let resolved = options;
+        if (!resolved.ref) {
```

Comment:
If this is an enhancement to calculate the `ref` based on the currently active widget, we could get rid of [the corresponding code](https://github.com/theia-ide/theia/pull/3451/files#diff-a40e16dd2fcfab480a9e0a4e17af95ebL246) in the Preview. What do you think? A follow-up task would be sufficient.

CC: @AlexTugarev

Reply:
@kittaakos, I'd really like to move the "open to the side" logic to the shell.
 1. if there is an area next (left or right, dep. on the option) to current widget, use `tab-after` in that area.
 2. otherwise split this area. 

but for that task the resolution of options should look different, I believe. for instance the mode is not changed.

Reply:
OK.